Tasks/Common/azurerest-common/azurerestutility.ts

```diff
@@ -298,16 +344,21 @@ export async function swapWebAppSlot(SPN, resourceGroupName: string, webAppName:
     );
 
     tl._writeLine(tl.loc('StartingSwapSlot',webAppName));
-    httpObj.send('POST', url, body, headers, (error, response, body) => {
+    httpObj.send('POST', url, body, headers, async (error, response, body) => {
         if(error) {
             deferred.reject(error);
         }
-        if(response.statusCode === 202) {
-            deferred.resolve(tl.loc("Successfullyswappedslots", webAppName, sourceSlot, targetSlot));
+        else if(response.statusCode === 202) {
+            await monitorSlotSwap(SPN, webAppName, resourceGroupName, sourceSlot, targetSlot, response.headers.location).then(() => {
+                deferred.resolve(tl.loc("Successfullyswappedslots", webAppName, sourceSlot, targetSlot));
+            }).catch((error) => {
+                tl.warning(error);
+                deferred.resolve(tl.loc("Successfullyswappedslots", webAppName, sourceSlot, targetSlot));
```

Comment:
done. if operation status is not completed, task will end successfully with warning message.